src/transformers/trainer.py

```diff
@@ -3842,13 +3843,15 @@ def evaluation_loop(
         if args.past_index >= 0:
             self._past = None
 
-        # Initialize containers
-        all_losses = EvalLoopContainer(self.args.eval_do_concat_batches, padding_index=-100)
-        all_preds = EvalLoopContainer(self.args.eval_do_concat_batches, padding_index=-100)
-        all_labels = EvalLoopContainer(self.args.eval_do_concat_batches, padding_index=-100)
-        all_inputs = EvalLoopContainer(self.args.eval_do_concat_batches, padding_index=-100)
+        metrics = {}
+        is_evaluation_stage = description == "Evaluation"
+        do_accumulate_epoch_data = not (is_evaluation_stage and args.batch_eval_metrics)
```

Comment:
can you add some comments to explain in which case we accumulate the data ?

Reply:
Added in https://github.com/huggingface/transformers/pull/31388/commits/7d18eb67a5541719b9ea829012927f288df645b8